src/Crypto/Polyfill/ByteArray.php

```diff
@@ -0,0 +1,265 @@
+                foreach ($tmp as $t) {
+                    if (strlen($t) < 1) {
+                        continue;
+                    }
+                    $arr []= \unpack('C', $t)[1] & 0xff;
+                }
+            }
+            $size = \count($arr);
+        } elseif ($size instanceof ByteArray) {
+            $arr = $size->toArray();
+            $size = $size->count();
+        } elseif (!\is_int($size)) {
+            throw new \InvalidArgumentException(
+                'Argument must be an integer, string, or array of integers.'
+            );
+        }
+
+        parent::__construct($size);
+
+        if (!empty($arr)) {
+            // Populate this object with values from constructor argument
+            foreach ($arr as $i => $v) {
+                $this->offsetSet($i, $v);
+            }
+        } else {
+            // Initialize to zero.
+            for ($i = 0; $i < $size; ++$i) {
+                $this->offsetSet($i, 0);
+            }
+        }
+    }
+
+    /**
+     * Encode an integer into a byte array. 32-bit (unsigned), big endian byte order.
+     *
+     * @param int $num
+     * @return self
+     */
+    public static function enc32be($num)
+    {
+        return new ByteArray(\pack('N', $num));
+    }
+
+    /**
+     * @param ByteArray $other
+     * @return bool
+     */
+    public function equals(ByteArray $other)
+    {
+        if ($this->count() !== $other->count()) {
+            return false;
+        }
+        $d = 0;
+        for ($i = $this->count() - 1; $i >= 0; --$i) {
+            $d |= $this[$i] ^ $other[$i];
+        }
+        return $d === 0;
+    }
+
+    /**
+     * @param ByteArray $array
+     * @return ByteArray
+     */
+    public function exclusiveOr(ByteArray $array)
+    {
+        self::needs(
+            $this->count() === $array->count(),
+            'Both ByteArrays must be equal size for exclusiveOr()'
+        );
+        $out = clone $this;
+        for ($i = 0; $i < $this->count(); ++$i) {
+            $out[$i] = $array[$i] ^ $out[$i];
+        }
+        return $out;
+    }
+
+    /**
+     * Returns a new ByteArray incremented by 1 (big endian byte order).
+     *
+     * @param int $increase
+     * @return self
+     */
+    public function getIncremented($increase = 1)
+    {
+        $clone = clone $this;
+        $index = $clone->count();
+        while ($index > 0) {
+            --$index;
+            $tmp = ($clone[$index] + $increase) & PHP_INT_MAX;
+            $clone[$index] = $tmp & 0xff;
+            $increase = $tmp >> 8;
+        }
+        return $clone;
+    }
+
+    /**
+     * Sets a value. See SplFixedArray for more.
+     *
+     * @param int $index
+     * @param int $newval
+     * @return void
+     */
+    public function offsetSet($index, $newval)
+    {
+        try {
+            parent::offsetSet($index, $newval & 0xff);
+        } catch (\Exception $ex) {
+            var_dump([$index, $newval]);
```

Comment:
Should there be a test to cover this exception in `ByteArrayTest`?

Reply:
It should be removed. I'll fix that in the next commit.